src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -75,11 +79,46 @@ void handle_reshape::run(program& p) {
             // vector for storing nodes that are reorder type, for which splitted primitives are needed (except for the
             // first one where orginal reshape will be used)
             std::vector<program_node*> reorder_node_to_split;
+            std::vector<program_node*> onednn_users;
 
             // find the users of reshape that are reorder type, if none present then skip the current node
+            // find onednn users
             for (const auto& user : node->get_users()) {
                 if (user->is_type<reorder>())
                     reorder_node_to_split.push_back(user);
+                if (user->get_preferred_impl_type() == cldnn::impl_types::onednn)
+                    onednn_users.push_back(user);
+            }
+
+            // If onednn user doesn't support new input data type, remove target reorder_node
+            if (!onednn_users.empty() && !reorder_node_to_split.empty()) {
```

Comment:
Could you add more comments why the siblings of reorder should be checked?

Reply:
Added more comments. Thanks!